frontend/src/components/BCDataGrid/BCDataGridEditor.jsx

```diff
@@ -0,0 +1,205 @@
+} from '@/components/BCDataGrid/components'
+import { v4 as uuid } from 'uuid'
+import PropTypes from 'prop-types'
+
+/**
+ * BCDataGridEditor component is a customizable data grid editor using ag-grid.
+ * It provides features like adding new rows, deleting selected rows, and saving changes.
+ *
+ * @param {function} saveData - Function to save changes made to the grid.
+ * @param {function} onGridReady - Callback function invoked when the grid is initialized and ready.
+ * @param {object} gridApi - Reference to the ag-grid API.
+ * @param {object} columnApi - Reference to the ag-grid column API.
+ * @param {array} rowData - Data to be displayed in the grid.
+ * @param {string} gridKey - Unique key identifier for the grid.
+ * @param {function} getRowNodeId - Function to get the unique identifier for each row.
+ * @param {object} gridRef - Reference to the ag-grid instance.
+ * @param {array} columnDefs - Definitions for the columns in the grid.
+ * @param {object} defaultColDef - Default column definition for the grid.
+ * @param {string} highlightedRowId - ID of the row to be highlighted.
+ */
+function BCDataGridEditor({
+  saveData,
+  onGridReady,
+  gridApi,
+  columnApi,
+  rowData,
+  gridKey,
+  getRowNodeId,
+  gridRef,
+  columnDefs,
+  defaultColDef,
+  highlightedRowId,
+  className,
+  ...others
+}) {
+  // Define framework components for ag-grid
+  const frameworkComponents = {
+    simpleEditor: SimpleEditor,
+    asyncValidationEditor: AsyncValidationEditor,
+    autocompleteEditor: AutocompleteEditor,
+    dateEditor: DateEditor,
+    actionsRenderer: ActionsRenderer,
+    aysncSuggestionEditor: AysncSuggestionEditor
+  }
+
+  // Default ag-grid options
+  const defaultGridOptions = useMemo(
+    () => ({
+      undoRedoCellEditing: true,
+      undoRedoCellEditingLimit: 5,
+      reactiveCustomComponents: true,
+      overlayNoRowsTemplate: 'No rows found',
+      autoSizeStrategy: { type: 'fitCellContents' },
+      suppressDragLeaveHidesColumns: true,
+      suppressMovableColumns: true,
+      suppressColumnMoveAnimation: false,
+      rowSelection: 'multiple',
+      rowHeight: 45,
+      headerHeight: 40,
+      animateRows: true,
+      suppressPaginationPanel: true,
+      suppressScrollOnNewData: true,
+      suppressCsvExport: false,
+      components: frameworkComponents,
+      onFirstDataRendered: (params) => {
+        params.api?.setFocusedCell(0, params.api?.getColumns()[2].getColId())
+
+        params.api?.startEditingCell({
+          rowIndex: 0,
+          colKey: params.api?.getColumns()[2].getColId()
+        })
+      },
+      // stopEditingWhenCellsLoseFocus: true,
+      getRowStyle: highlightedRowId
+        ? (params) => {
+            if (params.node.id === highlightedRowId) {
+              return { backgroundColor: '#fade81' }
+            }
+          }
+        : undefined
+    }),
+    [highlightedRowId]
+  )
+
+  // Function to add a new row
+  const addRow = useCallback(() => {
```

Comment:
We need some error handling on these action methods

```jsx
const addRow = useCallback(() => {
  try {
    const id = uuid()
    const emptyRow = { id, modified: true }
    gridApi.applyTransaction({ add: [emptyRow] })
  } catch (error) {
    console.error('Error adding row:', error)
    // Handle error gracefully (e.g., show a notification to the user)
  }
}, [gridApi])

const deleteRow = useCallback(() => {
  try {
    const selectedData = columnApi?.api?.getSelectedRows()
    gridApi?.applyTransaction({ remove: selectedData })
  } catch (error) {
    console.error('Error deleting row:', error)
    // Handle error gracefully (e.g., show a notification to the user)
  }
}, [columnApi, gridApi])

Reply:
As part of [#432](https://github.com/bcgov/lcfs/issues/432), I'll be moving these methods to AddFuelCode component and also implement react Query to make post calls as required. This was created for demo purpose only.